repos/system_upgrade/el7toel8/actors/kernel/checkinstalledkernels/libraries/checkinstalledkernels.py

```diff
@@ -1,77 +1,104 @@
+    # based OSs, as the rpm python module is available here just for Python2
+    # - and vice versa on F31+
+    # This will not happen in real run, just in case of unit-tests..
+    def labelCompare(*args):
+        raise NotImplementedError(
+            "The labelCompare function is not implemented for the python"
+            " you are using."
+        )
+
 from leapp import reporting
 from leapp.exceptions import StopActorExecutionError
-from leapp.libraries.common.config import architecture
+from leapp.libraries.common.config import architecture, version
 from leapp.libraries.stdlib import api
 from leapp.models import InstalledRedHatSignedRPM
 
 
-def _normalize_version(version):
-    if len(version) != 3:
-        if len(version) > 3:
-            api.current_logger().debug('Version {} has more than three components, trimming'.format(version))
-            del version[3:]
-        elif len(version) < 3:
-            api.current_logger().debug('Version {} has less than three components, padding'.format(version))
-            while len(version) < 3:
-                version.append(0)
-        api.current_logger().debug('Normalised version to {}'.format(version))
-    return version  # could be omitted but it's useful for nesting calls
+def get_current_kernel_version():
+    """
+    Get the version of the running kernel as a string.
+    """
+    return api.current_actor().configuration.kernel.split('-')[0]
 
 
-def get_current_kernel_version():
+def get_current_kernel_release():
     """
-    Get the version of the running kernel as a tuple of three integers.
+    Get the release of the current kernel as a string.
     """
-    kernel = api.current_actor().configuration.kernel
-    return tuple(_normalize_version(list(
-        map(int, kernel.split('-')[0].split('.'))
-    )))
+    return api.current_actor().configuration.kernel.split('-')[1]
 
 
-def get_kernel_rpm_version(rpm):
+def get_current_kernel_evr():
     """
-    Get the version of a kernel RPM as a tuple of three integers.
+    Get a 3-tuple (EVR) of the current booted kernel.
 
-    :param rpm: An instance of an RPM derived model.
+    Epoch in this case is always empty string. In case of kernel, epoch is
+    never expected to be set.
     """
-    return tuple(_normalize_version(list(
-        map(int, rpm.version.split('.'))
-    )))
+    return ('', get_current_kernel_version(), get_current_kernel_release())
 
 
-def get_current_kernel_release():
+def get_pkgs(pkg_name):
     """
-    Get the release of the running kernel as an integer.
+    Get all installed packages of the given name signed by Red Hat.
     """
-    kernel = api.current_actor().configuration.kernel
-    return int(kernel.split('-')[1].split('.')[0])
+    rpms = next(api.consume(InstalledRedHatSignedRPM), InstalledRedHatSignedRPM()).items
+    return [pkg for pkg in rpms if pkg.name == pkg_name]
 
 
-def get_kernel_rpm_release(rpm):
+def get_EVR(pkg):
```

Comment:
Let's be consistent with uppercase and lowercase with the EVR in function names...